kernel/sleep/src/lib.rs

```diff
@@ -88,7 +88,7 @@ fn add_to_delayed_tasklist(new_node: SleepingTaskNode) {
 fn remove_next_task_from_delayed_tasklist() {
     let mut delayed_tasklist = DELAYED_TASKLIST.lock();
     if let Some(SleepingTaskNode { taskref, .. }) = delayed_tasklist.pop() {
-        taskref.unblock();
+        let _ = taskref.unblock();
```

Comment:
see top-level review comment about forwarding `Result`s instead of `unwrap`ping. This however is probably the least good option since it just silently masks failures.

Reply:
should we log something here? is a task being killed while it is sleeping a failure?

Reply:
yep, it's worth logging and then unwrapping. The whole `sleep` crate will be redesigned once we have better sync primitives, e.g., the ones you're working on as part of `std` stuff.

Another eventual goal is to force a killed task to unwind (and thus, release its resources) rather than just stopping its execution immediately. That'd be relevant here too, but for now I think an `unwrap` call is warranted.